ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -172,3 +173,62 @@ let get_host_pcis () =
 let igd_is_whitelisted ~__context pci =
   let vendor_id = Db.PCI.get_vendor_id ~__context ~self:pci in
   List.mem vendor_id !Xapi_globs.igd_passthru_vendor_whitelist
+
+let is_pci_hidden_cmdline ~__context ~self =
+  let cmdline =
+    match Unixext.read_lines ~path:"/proc/cmdline" with
+    | [x] ->
+        x
+    | _ ->
+        failwith "Unable to read cmdline"
+  in
+  let device = Db.PCI.get_pci_id ~__context ~self in
+  let elems = String.split_on_char ' ' cmdline in
+
+  let pciback =
+    List.find_opt
+      (fun s -> String.starts_with ~prefix:"xen-pciback.hide=" s)
+      elems
+  in
+  (* Look for the device id in the list of hidden devices
+   * pciback looks like: "xen-pciback.hide==(<id1>)(<id2>)..." *)
+  let contains str substr = Astring.String.is_infix ~affix:substr str in
+  match pciback with None -> false | Some value -> contains value device
+
+let determine_dom0_access_status ~__context ~self =
```

Comment:
The name commonly used is `get_dom0_access_status`

Reply:
It's more than just a get of a db field so I though determine was more accurate but I can change if necessary.